core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -226,14 +226,17 @@ protected String handleRequest(Channel channel, Marker marker, StreamInput buffe
         transportServiceAdapter.onRequestReceived(requestId, action);
         NettyTransportChannel transportChannel = null;
         try {
-            transport.inFlightRequestsBreaker().addEstimateBytesAndMaybeBreak(messageLengthBytes, "<transport_request>");
-            transportChannel = new NettyTransportChannel(transport, transportServiceAdapter, action, channel,
-                requestId, version, profileName, messageLengthBytes);
-
             final RequestHandlerRegistry reg = transportServiceAdapter.getRequestHandler(action);
             if (reg == null) {
                 throw new ActionNotFoundTransportException(action);
             }
+            long reservedBytes = 0;
+            if (reg.isCheckSizeLimit()) {
```

Comment:
That's correct. Where would you include bytes that come from requests that are excluded from the limit check and what would you do with them then? The circuit breaker API provides a method `addWithoutBreaking(long)` but this still counts all added bytes in the same internal data structure.